src/main/scala/io/qbeast/spark/internal/sources/v2/QbeastScanBuilder.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.internal.sources.v2
+
+import org.apache.spark.sql.connector.read.{Scan, ScanBuilder}
+import org.apache.spark.sql.execution.datasources.FileIndex
+import org.apache.spark.sql.types.StructType
+
+/**
+ * Implementation of org.apache.spark.sql.connector.read.ScanBuilder.
+ */
+class QbeastScanBuilder private[v2] (
```

Comment:
This scan builder should support at least `SupportsPushDownFilters` and `SupportsPushDownRequiredColumns`.